### ISSUE TITLE
Improving Transaction Details on BVGO

### ISSUE THREAD

**Issue by krisnaBukitVista** [Bukit-Vista/roadmap#117]

**Comment by Vidiskiu** [comment]
@katibpasha Evaluation by Friday

**Comment by katibpasha** [comment]
# Final Evaluation

This project was successfully developed according to the initial plan, aiming to provide better financial visibility for strategic partners. Below are the notes for each target item in this milestone:

### Add Payment Due Date/Estimasi Dana Diterima

![Image](https://github.com/user-attachments/assets/e7957226-5d0b-45d1-a3e5-8644341174e6)

![Image](https://github.com/user-attachments/assets/ef2d53d2-548c-4abe-8dde-a5345366c948)

### Rename Payment Status

![Image](https://github.com/user-attachments/assets/f04c3af0-d2b0-41af-b1e0-9f5188ba72a5)

### Adjust Logic for Received Money

```
SELECT *, property.bv_nett - property.payout_bv AS pending_bv, property.owner_nett - property.payout_owner AS pending_owner, property.payout_bv / property.bv_nett * 100 AS percentage_bv, property.payout_owner / property.owner_nett * 100 AS percentage_owner FROM (SELECT COUNT(booking.booking_id) AS total_booking, SUM(booking.booking_earned) AS total_earned, SUM(booking.booking_earned * (1 - IFNULL(property_contract.revenue_share_bv, 0.2))) AS owner_nett, SUM(booking.booking_earned * IFNULL(property_contract.revenue_share_bv, 0.2)) AS bv_nett, SUM(payout.payout_owner) AS payout_owner, SUM(payout.payout_bv) AS payout_bv FROM property LEFT JOIN booking ON booking.property_id = property.property_id LEFT JOIN currency ON currency.currency_code = booking.booking_currency LEFT JOIN (
                    SELECT
                    *,
                        LAG(property_contract.contract_signing_date, 1)
                        OVER (PARTITION BY property_id ORDER BY property_contract.contract_signing_date DESC) contract_end_date
                    FROM (
                        SELECT
                            property_contract.contract_id,
                            property_contract.property_id,
                            property_contract.revenue_share_bv,
                            IF(property_contract.contract_signing_date IS NOT NULL, property_contract.contract_signing_date, property_contract.created_at) AS contract_signing_date,
                            property_contract.created_at
                        FROM
                            property_contract
                    ) property_contract
                ) property_contract ON property_contract.property_id = property.property_id AND ((booking.booking_received_timestamp BETWEEN DATE(property_contract.contract_signing_date) AND DATE(property_contract.contract_end_date)) OR (booking.booking_received_timestamp >= DATE(property_contract.contract_signing_date) AND property_contract.contract_end_date IS NULL)) LEFT JOIN (
                    SELECT
                        payout.payout_id,
                        payout.booking_id,
                        SUM(
                            CASE
                                WHEN bank_details.role = 1 THEN payout.payout_amount
                                ELSE 0
                            END
                        ) AS payout_owner,
                        SUM(
                            CASE
                                WHEN bank_details.role = 2 THEN payout.payout_amount
                                ELSE 0
                            END
                        ) AS payout_bv,
                        bank_details_role.bank_role_id,
                        bank_details_role.bank_role
                    FROM
                        payout
                        LEFT JOIN bank_details ON payout.destination_bank_account_id = bank_details.bank_details_id
                        LEFT JOIN bank_details_role ON bank_details.role = bank_details_role.bank_role_id
                    GROUP BY
                        payout.booking_id
                ) payout ON payout.booking_id = booking.booking_id WHERE property.property_id = 'PR0924' AND booking.booking_check_in BETWEEN '2024-12-01' AND '2024-12-31') property
```
### Add 'Payment Sent' Status

![Image](https://github.com/user-attachments/assets/8e92d3cf-8445-4eb8-ad03-b4c57a2a95e5)

**Comment by katibpasha** [comment]
# SLA Result

Based on our hermes log monitoring start from Jan 14th 2025 (when transaction details improvement released) till today (Jan 17th 2025), there is no inquiry from strategic partner related to financial issue.

![Image](https://github.com/user-attachments/assets/8cf1a387-8b3f-41ae-8473-ade86c80acce)

Compared to a week before Jan 1st 2025 - Jan 13th 2025, there are four inquiries from partners related to financial issues:

![Image](https://github.com/user-attachments/assets/b86dac01-7202-4980-82ea-6df6c7134920)

![Image](https://github.com/user-attachments/assets/5ef836eb-d584-4490-a29c-8f1e3a2600b5)

![Image](https://github.com/user-attachments/assets/e260d067-182b-4ea1-9322-2a856dc6f22e)

![Image](https://github.com/user-attachments/assets/754a4659-db55-46f2-b226-0a4f79bbbcb0)

In conclusion, there has been a significant reduction in the number of escalation messages from partners related to finance. This decrease has even reached 100%, dropping from 4 inquiries to 0.

**Comment by Vidiskiu** [comment]
Overall Point: 5.5
Functional Complexity: 1.5

> The issue involves enhancing payment statuses with added labels and dates, along with new logic for payment receipt. It needs modifications to existing functionalities and adding new minor features, which introduces moderate functional complexity.

Technical Complexity: 1.5

> Technical effort includes adding new statuses and modifying logic, however, does not involve deep architectural changes or integrating with new APIs. Thus the technical complexity is seen as moderate.

UI/UX Complexity and Impact: 1.25

> Significant changes to UI/UX as it involves adding new labels and status messages which changes the flow and visual representation of transaction details. This could have a noticeable impact on user experience.

Testing and Quality Assurance: 0.75

> Requires thorough testing to ensure the updated statuses and logic reflect accurately in the UI without affecting the actual transaction process, but does not involve extensive cross-platform or performance testing.

Risk and Dependencies: 0.5

> Risk is low to moderate relating to ensuring the UI changes accurately represent the correct payment process, with a dependency on language selection for labels which is straightforward.
